### DISCUSSION TITLE
How not to fetch all the user data when using auth API?

### DISCUSSION THREAD

**Discussion by hereAlexT** [supabase/supabase#19215] [Questions]
As the screenshot below, all the user-related information is fetched including the date time of `confirmed_at`, `created_at`, and `last_sign_in_at`.

I would like to use these data for user identity confirmation. For example, when a user loses his password, he can use this information to get the password back. So I want this column not to be fetched **on the server side**, any possible solution?

Thank you!

![image](https://github.com/supabase/supabase/assets/79433869/9542f4d8-e59a-4e60-ae9b-22b0aa57a56d)

Code:
```typescript
  const { data: { user, session }, error } = await supabase.auth.signInWithPassword({
            email: email,
            password: password
        }
```

**Comment by GaryAustin1** [comment]
It is not possible to change what the user object returns.

The only two fields you can ADD to are app_metadata and user_metadata.

**Reply by hereAlexT** [reply]
Is it possible to create an edge function in between to eliminate some columns and then return to the user?

**Reply by GaryAustin1** [reply]
No.  You can't change the auth schema or session or user object.  That is all handled in the Supabase client and auth calls.  There is no intercepting those except the insert/update/delete on auth.users to do things like a profile table.
You need to come up with something else to store in app_metadata or your own profile table.
